squid/customize.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+#
+# This file provides a parametrized version of
+#  /etc/squid/customize.sh
+# that comes from the RPM.
+# Note:
+#   It will not be overwritten by upgrades.
+# See customhelps.awk for information on predefined edit functions.
+# Avoid single quotes in the awk source or you have to protect them from bash.
+#
+
+# If some of the variables are not defined, set the to their default values
+# These values should reflect what was in the RPM
+
+if [ -z "$SQUID_IPRANGE" ]; then
+  export SQUID_IPRANGE="10.0.0.0/8 172.16.0.0/12 192.168.0.0/16 fc00::/7 fe80::/10" 
+fi
+
+if [ -z "$SQUID_CACHE_MEM" ]; then
+  export SQUID_CACHE_MEM="128 MB" 
+fi
+
+if [ -z "$SQUID_CACHE_DISK" ]; then
+  export SQUID_CACHE_DISK="10000"
+fi
```

Comment:
Can we just set the following in the Dockerfile?

```
ENV SQUID_IPRANGE="10.0.0.0/8 172.16.0.0/12 192.168.0.0/16 fc00::/7 fe80::/10" 
ENV SQUID_CACHE_MEM="128 MB"
ENV SQUID_CACHE_DISK="10000"
```

Then the above comment about default values can be moved to the Dockerfile, too

Reply:
Having the test in the script is still useful; you don't want to operate on undefined variables.

But I could add it to the Dockerfile, too.

Reply:
See 214f380

Reply:
If we have defaults in the Dockerfile, then they'll only be undefined if the user unsets them. If that's happening, it's most likely a mistake on the user's part, so I don't think asserting defaults is the right thing to do.

Reply:
The alternative is a hard fail... Are you OK with that?
I can go with it.

Reply:
I think a hard fail with an error message indicating which vars are unset is the appropriate behavior. So if we replace the export lines with echo'ing error messages and non-zero exits, would that show up in docker logs? I'm guessing not so perhaps the env var validation should go in the supervisord startup script.